cmd/thv/app/config.go

```diff
@@ -113,7 +113,7 @@ func autoDiscoveryCmdFunc(_ *cobra.Command, args []string) error {
 		// If auto-discovery is enabled, update all registered clients with currently running MCPs
 		if enabled && len(c.Clients.RegisteredClients) > 0 {
 			for _, clientName := range c.Clients.RegisteredClients {
-				if err := addRunningMCPsToClient(clientName); err != nil {
+				if err := addRunningMCPsToClient(cmd.Context(), clientName); err != nil {
```

Comment:
Not related to the main point of the PR, but I removed the manual creation of context and reused the Cobra command's context instead.